### PR TITLE
modify Trimmomatic command; allow small insert-size (rev. complement) reads through QC

### PR DESCRIPTION
**Pull request by katrinakalantar**
# Description

Several users have recently had issues with samples where % Pass QC is low ( < 25% ). Many reads were being removed during the trimmomatic step of QC due to trimmomatic default behavior to remove R2 reads that are the reverse complement of the R1, thus making them "orphan reads". These orphaned reads were then being filtered out. We would rather keep both R1 and R2 in these cases as they still provide useful information (effectively comparable to a SE read). Additionally, the MINLEN parameter has been reduced to 35 given that we enable L filtering downstream.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. Trimmomatic was run manually using the pre-existing command (results below).

`java -jar ~/Documents/TOOLS/Trimmomatic-0.39/trimmomatic-0.39.jar PE -phred33 -trimlog logfile -summary statsummary unmapped1.fastq.gz unmapped2.fastq.gz unmapped_paired_out1.fq.gz unmapped_unpaired_out1.fq.gz unmapped_paired_out2.fq.gz unmapped_unpaired_out2.fq.gz ILLUMINACLIP:illumina_TruSeq3-PE-2_NexteraPE-PE.fasta:2:30:10 MINLEN:75`

`Input Read Pairs: 20510 Both Surviving: 3900 (19.02%) Forward Only Surviving: 10966 (53.47%) Reverse Only Surviving: 32 (0.16%) Dropped: 5612 (27.36%)`

2. The command was updated and re-run manually (results below) and the results were compared.

`java -jar ~/Documents/TOOLS/Trimmomatic-0.39/trimmomatic-0.39.jar PE -phred33 -trimlog logfile -summary statsummary unmapped1.fastq.gz unmapped2.fastq.gz unmapped_paired_out1.fq.gz unmapped_unpaired_out1.fq.gz unmapped_paired_out2.fq.gz unmapped_unpaired_out2.fq.gz ILLUMINACLIP:illumina_TruSeq3-PE-2_NexteraPE-PE.fasta:2:30:10:8:true MINLEN:75`

`Input Read Pairs: 20510 Both Surviving: 14862 (72.46%) Forward Only Surviving: 4 (0.02%) Reverse Only Surviving: 32 (0.16%) Dropped: 5612 (27.36%)`

3. I used this branch to re-run all the samples in project Rapid Response RR025, and the % Passed QC increased from ~5-10% to ~85-90%.

## Impacted Areas in Application

List general components of the application that this PR will affect:

- QC metrics

# Checklist:

- [ ] I have run through the testing script to make sure current functionality is unchanged
- [x] I have done relevant tests that prove my fix is effective or that my feature works
- [x] I have spent time testing out edge cases for my feature
- [ ] I have updated the test script or pull request template if necessary
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by cdebourcy** [comment]
Yay!

Please up the second digit in the version (3.6.5 --> 3.7.0) to ensure reruns are not evaluated lazily: https://github.com/chanzuckerberg/idseq-dag/blob/master/idseq_dag/__init__.py#L2

Also, please add a corresponding description of the change to the release notes:
https://github.com/chanzuckerberg/idseq-dag/blob/master/README.md

**Comment by tfrcarvalho** [comment]
> Please up the second digit in the version (3.6.5 --> 3.7.0) to ensure reruns are not evaluated lazily: [/idseq_dag/__init__.py@`master`#L2]

@cdebourcy I guess this means that we will no longer able to do the easy rerun describe in the oncall doc to enable coverage viz if users ask? or if we can set a specific version, should we update the oncall doc or do it just rerun all at 3.7?

**Comment by cdebourcy** [comment]
> > Please up the second digit in the version (3.6.5 --> 3.7.0) to ensure reruns are not evaluated lazily: [/idseq_dag/**init**.py@`master`#L2]
> 
> @cdebourcy I guess this means that we will no longer able to do the easy rerun described in the oncall doc to enable coverage viz if users ask? or if we can set a specific version, should we update the oncall doc or do it just rerun all at 3.7?

Hmm good question. I guess we will have to lose that flexibility *eventually*... we should coordinate with @eggerdesigns on a rollout plan. We should also announce the trimmomatic change, because it has the potential to drastically change results (the previous behavior is pretty much a bug).

**Comment by katrinakalantar** [comment]
I wonder if there is a better way to roll this out - bundled with some change to auto-verify the insert size so that users can determine whether their run would benefit from being re-run with this version.

**Comment by katrinakalantar** [comment]
Validation on Medical Detectives cases: the results after the trimmomatic parameter update are tightly correlated to the original results, but have a slightly higher total number of reads (a function of allowing more reads through QC). For each sample tested/plotted below, the total number of reads passing QC was increased... Pt 006 = +11%, Pt 002 = +45%, Pt 001 = +17%, Pt 016 = +22%.

image

**Comment by cdebourcy** [comment]
Are the top organisms in the reports for the samples you tested still the same and correct?

**Comment by katrinakalantar** [comment]
> Are the top organisms in the reports for the samples you tested still the same and correct?

Yes, correct.

File: idseq_dag/__init__.py [hunk 1]
```diff
@@ -1,2 +1,4 @@
 ''' idseq_dag '''
-__version__ = "3.6.6"
+
```
Comment (cdebourcy) [inline idseq_dag/__init__.py]: remove blank line?
